### PR TITLE
gh-101100: Fix sphinx warnings in `howto/*`

### PR DESCRIPTION
**Pull request by ghost**
Removed sphinx roles of some classes and methods, as the classes are defined in the sample codes, not in Python.

Fixed warnings:
```
howto/descriptor.rst:45: WARNING: py:class reference target not found: Ten [ref.class]
howto/descriptor.rst:218: WARNING: py:class reference target not found: Person [ref.class]
howto/descriptor.rst:218: WARNING: py:class reference target not found: Person [ref.class]
howto/descriptor.rst:256: WARNING: py:class reference target not found: Person [ref.class]
howto/descriptor.rst:340: WARNING: py:class reference target not found: Validator [ref.class]
howto/descriptor.rst:363: WARNING: py:class reference target not found: Validator [ref.class]
howto/descriptor.rst:363: WARNING: py:meth reference target not found: validate [ref.meth]
howto/descriptor.rst:372: WARNING: py:class reference target not found: OneOf [ref.class]
howto/descriptor.rst:374: WARNING: py:class reference target not found: Number [ref.class]
howto/descriptor.rst:378: WARNING: py:class reference target not found: String [ref.class]
howto/descriptor.rst:876: WARNING: py:class reference target not found: Field [ref.class]
howto/descriptor.rst:1143: WARNING: py:func reference target not found: Property [ref.func]
howto/enum.rst:78: WARNING: py:class reference target not found: Weekday [ref.class]
howto/enum.rst:90: WARNING: py:class reference target not found: Weekday [ref.class]
howto/enum.rst:113: WARNING: py:class reference target not found: Weekday [ref.class]
howto/enum.rst:131: WARNING: py:class reference target not found: Weekday [ref.class]
howto/enum.rst:576: WARNING: py:class reference target not found: Animal [ref.class]
howto/enum.rst:893: WARNING: py:class reference target not found: FloatEnum [ref.class]
```

* Issue: gh-101100

----
📚 Documentation preview 📚: https://cpython-previews--127084.org.readthedocs.build/

### CODE REVIEW

File: Doc/howto/descriptor.rst [hunk 1]
```diff
@@ -42,7 +42,7 @@ add new capabilities one by one.
 Simple example: A descriptor that returns a constant
 ----------------------------------------------------
 
-The :class:`Ten` class is a descriptor whose :meth:`__get__` method always
+The ``Ten`` class is a descriptor whose :meth:`__get__` method always
```
Comment (hugovk) [inline Doc/howto/descriptor.rst]: If we keep the `class` role but add an exclamation mark then we keep the formatting of a class but don't try to resolve the reference, and get no warning:

```suggestion
The :class:`!Ten` class is a descriptor whose :meth:`!__get__` method always
```

Re: https://devguide.python.org/documentation/markup/#roles

Same applies to the others in this PR.

And we can also do it for the example `__get__` method here.

Comment (ghost) [reply Doc/howto/descriptor.rst]: @hugovk Fixed class references by using `!` prefix. `__get__` is documented in <https://docs.python.org/3/reference/datamodel.html#object.__get__> , so I think it is better to fix `__get__` to `~object.__get__`.

Comment (hugovk) [reply Doc/howto/descriptor.rst]: > `__get__` is documented in [docs.python.org/3/reference/datamodel.html#object.__get__](https://docs.python.org/3/reference/datamodel.html#object.__get__) , so I think it is better to fix `__get__` to `~object.__get__`.

Yes, good idea.

**Review by hugovk — APPROVED** [review]
Thanks!

**Comment by hugovk** [comment]
@koyuki7w Please could you take care of the 3.12 backport? Thanks!

**Comment by koyuki7w** [comment]
@hugovk Thanks. Opened 3.12 backport PR.

**Pull request merged**